### PR TITLE
Rename Azure.ResourceManager.StackHci to Azure.ResourceManager.Hci

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Azure.ResourceManager.Hci 1.0.0-beta.1 is released on the release_stackhci branch as the main branch now depends on an alpha version of Azure.ResourceManager. This PR merges the changes back to the main branch.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/azurestackhci/Azure.ResourceManager.Hci/README.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
-# Azure Azure.ResourceManager.StackHCI Management client library for .NET
+# Azure azurestackhci Management client library for .NET
```
Comment (Yao725) [inline sdk/azurestackhci/Azure.ResourceManager.Hci/README.md:1]: Maybe below is better?
```suggestion
# Azure Stack HCI Management client library for .NET
```

File: sdk/azurestackhci/Azure.ResourceManager.Hci/README.md [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 
 This package follows the [new Azure SDK guidelines](https://azure.github.io/azure-sdk/general_introduction.html) which provide a number of core capabilities that are shared amongst all Azure SDKs, including the intuitive Azure Identity library, an HTTP Pipeline with custom policies, error-handling, distributed tracing, and much more.
 
 ## Getting started 
 
 ### Install the package
 
-Install the Azure Azure.ResourceManager.StackHCI management library for .NET with [NuGet](https://www.nuget.org/):
+Install the Azure azurestackhci management library for .NET with [NuGet](https://www.nuget.org/):
```
Comment (Yao725) [inline sdk/azurestackhci/Azure.ResourceManager.Hci/README.md:9]: Same as above.

File: sdk/azurestackhci/Azure.ResourceManager.Hci/tests/azurestackhciManagementTestEnvironment.cs [hunk 3]
```diff
@@ -3,9 +3,9 @@
 
 using Azure.Core.TestFramework;
 
-namespace Azure.ResourceManager.StackHCI.Tests
+namespace Azure.ResourceManager.Hci.Tests
 {
-    public class StackHCIManagementTestEnvironment : TestEnvironment
+    public class azurestackhciManagementTestEnvironment : TestEnvironment
```
Comment (Yao725) [inline sdk/azurestackhci/Azure.ResourceManager.Hci/tests/azurestackhciManagementTestEnvironment.cs:8]: `a` should be capitalized.

**Comment by ArthurMa1978** [comment]
there are 2 files conflicts

File: sdk/azurestackhci/Azure.ResourceManager.Hci/src/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+
+[assembly: Azure.Core.AzureResourceProviderNamespace("azurestackhci")]
+
+[assembly: InternalsVisibleTo("Azure.ResourceManager.Hci.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100d15ddcb29688295338af4b7686603fe614abd555e09efba8fb88ee09e1f7b1ccaeed2e8f823fa9eef3fdd60217fc012ea67d2479751a0b8c087a4185541b851bd8b16f8d91b840e51b1cb0ba6fe647997e57429265e85ef62d565db50a69ae1647d54d7bd855e4db3d8a91510e5bcbd0edfbbecaa20a7bd9ae74593daa7b11b4")]
```
Comment (Yao725) [inline sdk/azurestackhci/Azure.ResourceManager.Hci/src/Properties/AssemblyInfo.cs:8]: Please add this line here to make the test proxy work as expected：
```
[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2, PublicKey=0024000004800000940000000602000000240000525341310004000001000100c547cac37abd99c8db225ef2f6c8a3602f3b3606cc9891605d02baa56104f4cfc0734aa39b93bf7852f7d9266654753cc297e7d2edfe0bac1cdcf9f717241550e0a7b191195b7667bb4f64bcb8e2121380fd1d9d46ad2d92d2d15605093924cceaf74c4861eff62abf69b9291ed0a340e113be11e6a7d3113e92484cf7045cc7")]
```

File: sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    transform: >
+      $.aadClientId.format = "uuid";
+      $.aadTenantId.format = "uuid";
+      $.cloudId.format = "uuid";
+  - from: clusters.json
+    where: $.definitions.ClusterPatchProperties.properties
+    transform: >
+      $.aadClientId.format = "uuid";
+      $.aadTenantId.format = "uuid";
+  - from: clusters.json
```
Comment (Yao725) [inline sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md:36]: I think it's better to use the `rename rule` configuration here.

**Review by Yao725 — APPROVED** [review]
Left some minor comments and the rest LGTM.

**Pull request merged**